code/API_definitions/device-reachability-status.yaml

```diff
@@ -0,0 +1,403 @@
+
+    Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation.
+
+    It is important to remark that in cases where personal user data is processed by the API, and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of 3-legged access tokens becomes mandatory. This measure ensures that the API remains in strict compliance with user privacy preferences and regulatory obligations, upholding the principles of transparency and user-centric data control.
+
+    ## Further info and support
+
+    (FAQs will be added in a later version of the documentation)
+
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: project-email@sample.com
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: wip
+externalDocs:
+  description: Product documentation at CAMARA
+  url: https://github.com/camaraproject/
+
+servers:
+  - url: "{apiRoot}/device-reachability-status/v0"
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root
+tags:
+  - name: Device reachability status
+    description: Operations to get the current reachability status of a device
+paths:
+  /retrieve:
+    post:
+      tags:
+        - Device reachability status
+      summary: "Get the current reachability status information"
+      description: Get the current reachability status information
+      operationId: getReachabilityStatus
+      parameters:
+        - $ref: '#/components/parameters/x-correlator'
+      security:
+        - openId:
+            - device-reachability-status:read
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/RequestReachabilityStatus"
+        required: true
+      responses:
+        "200":
+          description: Contains information about current reachability status
+          headers:
+            x-correlator:
+              $ref: '#/components/headers/x-correlator'
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ReachabilityStatusResponse"
+              examples:
+                Connected-With-SMS:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    reachabilityStatus: CONNECTED_SMS
+                Connected-With-DATA:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    reachabilityStatus: CONNECTED_DATA
+                Not-Connected:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"

```

Comment:
Do we have an example where network is connected to both SMS and Data  and we need to send a response accordingly like: {
  "lastStatusTime": "2024-02-20T10:41:38.657Z",
  "reachabilityStatus": "CONNECTED_SMS_DATA"
}

Reply:
For me no as my understanding DATA means that SMS is also available.
@sachinvodafone I will merge this PR so probably we could continue the conversation in Discussions tab  as it does not impact this merging.